gramps/plugins/graph/gvrelgraph.py

```diff
@@ -182,10 +182,71 @@ def write_report(self):
         self.person_handles = self._filter.apply(self.database,
                     self.database.iter_person_handles())
 
+        self.person_handles = self.sort_persons()
+
         if len(self.person_handles) > 1:
             self.add_persons_and_families()
             self.add_child_links_to_families()
 
+    def sort_persons(self):
+        "sort persons by close relations"
+
+        # first make a list of all persons who don't have any parents
+        rootNodes = list()
+        for person_handle in self.person_handles:
+            person = self.database.get_person_from_handle(person_handle)
+            if person is None:
+                continue
+            hasParent = False
+            for parent_handle in find_parents(self.database, person):
+                if not parent_handle in self.person_handles:
+                    continue
+                parent = self.database.get_person_from_handle(parent_handle)
+                if parent is None:
+                    continue
+                hasParent = True
+            if not hasParent:
+                rootNodes.append(person_handle)
+
+        # now start from all root nodes we found and traverse their trees
+        outlist = list()
+        p_done = {}
```

Comment:
I think you want to use a `set()` here instead of a dictionary. The actual value is unused.